### PR TITLE
Create unsupported bitfield implementation

### PR DESCRIPTION
**Pull request by jonahwilliams**
allows compiling flutter in environments without 64 bit ints.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/bitfield_unsupported.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Unsupported.
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: This doc comment is unhelpful.

File: packages/flutter/lib/src/foundation/bitfield_unsupported.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+int get kMaxUnsignedSMI => throw UnsupportedError('Not supported in environments without 64 bit ints');
+
+/// A BitField over an enum (or other class whose values implement "index").
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: nit: first sentence should be its own paragraph.

File: packages/flutter/lib/src/foundation/bitfield.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+// found in the LICENSE file.
+
+// BITFIELD
+
+/// The largest SMI value.
+///
+/// See <https://www.dartlang.org/articles/numeric-computation/#smis-and-mints>
+const int kMaxUnsignedSMI = 0x3FFFFFFFFFFFFFFF;
+
+/// A BitField over an enum (or other class whose values implement "index").
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield.dart]: nit: first sentence should be its own paragraph.

File: packages/flutter/lib/src/foundation/bitfield_unsupported.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  BitField(int _length)
+    : assert(_length <= 62);
+
+  /// Creates a bit field filled with a particular value.
+  ///
+  /// If the value argument is true, the bits are filled with ones. Otherwise,
+  /// the bits are filled with zeros.
+  ///
+  /// The given length must be at most 62.
+  BitField.filled(int length, bool _) // ignore: avoid_unused_constructor_parameters
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: please add explanation to all `// ignore: `, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#comment-all--ignores

File: packages/flutter/lib/src/foundation/bitfield_unsupported.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+int get kMaxUnsignedSMI => throw UnsupportedError('Not supported in environments without 64 bit ints');
+
+/// A BitField over an enum (or other class whose values implement "index").
+/// Only the first 62 values of the enum can be used as indices.
+class BitField<T extends dynamic> {
+  /// Creates a bit field of all zeros.
+  ///
+  /// The given length must be at most 62.
+  BitField(int _length)
+    : assert(_length <= 62);
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: Should the 62 be a const here and below like in the other class?

File: packages/flutter/lib/src/foundation/bitfield.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// BITFIELD
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield.dart]: This heading comment seems kind of redundant now that it has its own file?

File: packages/flutter/lib/src/foundation/bitfield_unsupported.dart (hunk 2 above)
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: Should the doc comments here and/or in the other implementation mention the conditions for when this thing is actually supported?

(I am also wondering how dartdocs deals with conditional imports? Does it only show the dart docs for one of the impls? If so, which one? Does this thing show up twice in dart docs now?)

Comment (jonahwilliams) [reply packages/flutter/lib/src/foundation/bitfield_unsupported.dart]: I don't think they should show up at all, but I'll need to test it.

Either the conditionally imported file has the same API, or we're in a world of hurt. That would mean all documentation should be on the main (non-conditional) implementation. I'm not sure if that means we should leave stub comments or something on the conditional file.

**Comment by jonahwilliams** [comment]
Turns out we don't need to do this quite yet

**Pull request closed without merging**